### ISSUE TITLE
Custom Windows System schemes make console unusable

### ISSUE THREAD

**Issue by DanRathbun** [Aerilius/sketchup-console-plus#17]
Custom Windows System schemes make console unusable.

Screenshot of console (v3.0.1) on Win7 with custom high contrast charcoal scheme:

![consolepluscharcoal](https://user-images.githubusercontent.com/334837/32196956-cfe535a8-bd98-11e7-9010-a15414204292.png)

**Comment by Aerilius** [comment]
The icons have been purposely not set to very dark colors so that they should be distinguishable from black background. I see that fails if the background is a similar gray.

It's probably not trivial to make images flip color (in contrast to fonts), unless I encode them in icon fonts, which will probably break on legacy systems that don't support webfonts.

**Comment by Aerilius** [comment]
Does commit 5dc2b6b55aafa11db20b3f2f167202f2c978aab7 in branch issue-dark-themes improve the situation?

**Comment by DanRathbun** [comment]
The manually downloaded master (v3.0.2) looks the same.

The best thing going forward would be to have a "themes" folder and beneath that separate named theme folders that have the css and images files in them. Along with a way for user to set the theme.
Then people could contribute theme folders to the project.

**Comment by Aerilius** [comment]
That branch is not in a release, only in the source repository.

Hmm, so far there were not yet a lot of people requesting this to justify themeability as a feature, but I'll wait and see. For now, I don't have a problem with users just overwriting the icon resources in the images folder. Or adding in the css file `.toolbar button img { filter: brightness(+200%); }`. But I cannot detect the actual color values of the OS theme, I cannot flip such a filter programmatically.

**Comment by DanRathbun** [comment]
> _Hmm, so far there were not yet a lot of people requesting this to justify themeability as a feature, but I'll wait and see._

Yeah, well it looks like I'm leaving SketchUp plugin development behind, so it will not affect me going forward.

**Comment by DanRathbun** [comment]
> _But I cannot detect the actual color values of the OS theme_ ...

https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Examples#Example_6:_getComputedStyle ?

https://developer.mozilla.org/en-US/docs/Web/API/CSS_Object_Model/Using_dynamic_styling_information

> _Or adding in the css file_ **`.toolbar button img { filter: brightness(+200%); }`**
... _I cannot flip such a filter programmatically._

```
rules = document.styleSheets[0].cssRules
for (var i = 0; i < rules.length-1; i++) { 
    if (rules[i].selectorText == ".toolbar button img") {
        rules[i].style.setProperty("filter", "brightness(100%)");  
        break;
    }
}
```
**?**
